augur/export_v2.py

```diff
@@ -1142,6 +1155,17 @@ def run(args):
             parse_node_data_and_metadata(T, node_data_file, metadata_file)
     config = get_config(args)
 
+    # Check additional metadata columns requested exist
+    additional_metadata_columns = []
+    if args.metadata_columns:
+        for col in args.metadata_columns:
+            # Match the column names corrected within parse_node_data_and_metadata
+            corrected_col = update_deprecated_names(col)
+            if corrected_col not in metadata_names:
+                print(f"WARNING: Requested metadata column {col!r} does not exist and will not be exported")
```

Comment:
Suggestion: Add test for this warning. Example:

```
Missing columns are skipped with a warning.

  $ ${AUGUR} export v2 \
  >  --tree tree.nwk \
  >  --metadata metadata.tsv \
  >  --metadata-columns "field_A" "missing_field" \
  >  --maintainers "Nextstrain Team" \
  >  --output dataset.json
  WARNING: Requested metadata column 'missing_field' does not exist and will not be exported
  …
```

Reply:
Added in b9b5649914d31786a57c928f9b67781209e43869